va/validation-authority.go

```diff
@@ -549,13 +551,26 @@ func (va *ValidationAuthorityImpl) validateChallenge(ctx context.Context, identi
 	}
 }
 
-// UpdateValidations runs the validate() method asynchronously using goroutines.
+// UpdateValidations runs the validate() method asynchronously using
+// goroutines.
+//
+// TODO(#1167): remove this method
 func (va *ValidationAuthorityImpl) UpdateValidations(authz core.Authorization, challengeIndex int) error {
 	// TODO(#1292): add a proper deadline here
-	go va.validate(context.TODO(), authz, challengeIndex)
+	go func(authz core.Authorization, challengeIndex int) {
+		ctx := context.Background()
+		va.RA.OnValidationUpdate(va.validate(ctx, authz, challengeIndex))
+	}(authz, challengeIndex)
 	return nil
 }
 
+// PerformValidation runs the validate() method synchronously and returns the
+// updated Challenge.
+func (va *ValidationAuthorityImpl) PerformValidation(authz core.Authorization, challengeIndex int) (core.Challenge, error) {
```

Comment:
My ideal API for this would just take a Challenge object and return `ValidationRecord, error`. I know a lot of the methods in VA expect to get an Authorization. This may be somewhat of a historical thing. Can you go through and see how practical it would be to make this take a Challenge instead of an Authorization and challengeIndex?